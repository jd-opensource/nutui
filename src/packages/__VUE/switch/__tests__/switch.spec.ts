import { mount, config } from '@vue/test-utils';
import Switch from '../index.vue';

beforeAll(() => {
  config.global.components = {};
});

afterAll(() => {
  config.global.components = {};
});

test('render ok', () => {
  const wrapper = mount(Switch);

  expect(wrapper.html()).toMatchSnapshot();
});

test('prop modelValue test', () => {
  const wrapper = mount(Switch, {
    props: {
      modelValue: true
    }
  });
  expect(wrapper.classes()).toContain('nut-switch-open');
});

test('prop disable test', () => {
  const wrapper = mount(Switch, {
    props: {
      disable: true
    }
  });
  expect(wrapper.classes()).toContain('nut-switch-disable');
});

test('prop loading test', () => {
  const wrapper = mount(Switch, {
    props: {
      loading: true
    }
  });
  expect(wrapper.find('.nut-icon-loading1')).toBeTruthy();
});

test('prop activeText test', () => {
  const wrapper = mount(Switch, {
    props: {
      activeText: 'test text'
    }
  });
  expect(wrapper.html()).toContain('test text');
});

test('prop activeValue test', () => {
  const wrapper = mount(Switch, {
    props: {
      modelValue: '0',
      activeText: '开',
      activeValue: '1',
      inactiveText: '关',
      inactiveValue: '0'
    }
  });
  expect(wrapper.find('.close').isVisible()).toBeTruthy();
});

test('emit click event', () => {
  const wrapper = mount(Switch);

  wrapper.trigger('click');
  expect(wrapper.emitted('change')).toBeTruthy();
});
