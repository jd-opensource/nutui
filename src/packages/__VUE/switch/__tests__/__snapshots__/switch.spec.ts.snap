// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render ok 1`] = `
"<view class="nut-switch nut-switch-close nut-switch-base">
  <view class="nut-switch-button">
    <!--v-if-->
    <!--v-if-->
  </view>
</view>"
`;
